### DISCUSSION TITLE
Can raster data be added to maps

### DISCUSSION THREAD

**Discussion by LukeParky** [heremaps/harp.gl#2353] [Q&A]
Hi, I have some raster files with depth values of water and want to add these to a map to show the data. Is there an established way of adding raster data to a map? Either in a 3D representation or as a colour mapping.

**Comment by LukeParky** [comment]
I have searched the documentation and examples, and raster data is mentioned but I can't find evidence that it is able to be used yet

**Reply by nzjony** [reply]
Have you seen: https://www.harp.gl/docs/master/examples/codebrowser.html?src=src%2Fdatasource_custom-texture-tile.ts ?

You just need to implement the method:
```
function getTexture(tile: Tile): Promise<[THREE.Texture, CopyrightInfo[]]> {
        return Promise.all([new TextureLoader().load("resources/wests_textures/clover.png"), []]);
}
```
    
I.e. assuming you know the way in which your textures are cut, you just need to use the correct tiling scheme and you can add a custom data provider to the `WebTileDataSource`. Is that what you mean?
